src/entities/sorted_state_adapter.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  EntityState,
+  IdSelector,
+  Comparer,
+  EntityStateAdapter,
+  Update,
+  EntityMap
+} from './models'
+import { createStateOperator } from './state_adapter'
+import { createUnsortedStateAdapter } from './unsorted_state_adapter'
+import { selectIdValue } from './utils'
+
+export function createSortedStateAdapter<T>(
+  selectId: IdSelector<T>,
+  sort: Comparer<T>
+): EntityStateAdapter<T>
+export function createSortedStateAdapter<T>(selectId: any, sort: any): any {
+  type R = EntityState<T>
```

Comment:
When you're done I'd try if we can type this somehow that we can get a Key type somewhere here and there's not so much confusion about number and string ids everywhere in the typings.

Reply:
Totally agreed.